finish/src/main/java/CreateTranscript.java

```diff
@@ -0,0 +1,175 @@
+           .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+           .setAccessType("offline")
+           .build();
+
+   LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8888).build();
+   return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+ }
+
+ /**
+  * Calls helper functions to create Doc, get audio file's transcript, and insert transcript into
+  * created Doc.
+  *
+  * @param {Object} service Docs authorization service to be able to use the Docs API.
+  */
+ private static void createTranscript(Docs service) throws IOException {
+   // Calls helper functions.
+   String docId = createDocument(service);
+   List<Request> toInsert = getTranscript();
+   insertText(service, toInsert, docId);
+ }
+
+ /**
+  * Creates a new Google Document. Once the document is created, returns its Document ID.
+  *
+  * @param {Object} service Docs authorized service to be able to create a Doc.
+  * @return {String} Returns the Document ID of the newly created Doc.
+  */
+ private static String createDocument(Docs service) throws IOException {
+   Document doc = new Document().setTitle("Transcript for " + AUDIO_FILE_PATH);
+   doc = service.documents().create(doc).execute();
+   String documentId = doc.getDocumentId();
+   return documentId;
+ }
+
+ /**
+  * Obtains the transcript of an audio file by calling the Google Speech-to-Text API.
+  *
+  * @return {Object} A list of requests of the audio file's transcript.
+  */
+ private static List<Request> getTranscript() throws IOException {
+   SpeechClient speech = SpeechClient.create();
+   Path path = Paths.get(AUDIO_FILE_PATH);
+   byte[] data = Files.readAllBytes(path);
+   ByteString audioBytes = ByteString.copyFrom(data);
+
+   // Configure request with local raw PCM audio.
+   RecognitionConfig config =
+       RecognitionConfig.newBuilder()
+           .setEncoding(AudioEncoding.LINEAR16)
+           .setLanguageCode("en-US")
+           .setSampleRateHertz(8000)
+           .build();
+   RecognitionAudio audio = RecognitionAudio.newBuilder().setContent(audioBytes).build();
+   RecognizeResponse response = speech.recognize(config, audio);
+   List<SpeechRecognitionResult> results = response.getResultsList();
+
+   // Initialize ArrayList to be filled with requests & returned.
+   List<Request> requests = new ArrayList<>();
+
+   for (SpeechRecognitionResult result : results) {
+     // Using the first + most likely alternative transcript
+     SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
+     String toInsert = alternative.getTranscript();
+
+     // Add requests array list to return.
+     requests.add(
+         new Request()
+             .setInsertText(
+                 new InsertTextRequest()
+                     .setText(toInsert)
+                     .setEndOfSegmentLocation(new EndOfSegmentLocation().setSegmentId(""))));
```

Comment:
do we need this if the id is ""?

Reply:
Yes: https://developers.google.com/docs/api/reference/rest/v1/documents/request#endofsegmentlocation

Reply:
definitely not intuitive